src/backend/renderer/gles2/mod.rs

```diff
@@ -622,6 +627,10 @@ unsafe fn link_program(
     let vert = compile_shader(gl, ffi::VERTEX_SHADER, vert_src)?;
     let frag = compile_shader(gl, ffi::FRAGMENT_SHADER, frag_src)?;
     let program = gl.CreateProgram();
```

Comment:
Unrelated to this pull request, but it creating the program fails, we end up leaking the vert and frag shaders. This is something we need to try to deal with in the gles renderer.

Reply:
could you explain me, how to free the shaders then I can add the free-management for it.

Reply:
so when you use the try operator (`?`), it will immediately return `Err` if there is a failure.

What needs to be done here is that if creating `frag` fails you need to delete the `vert` shader and then return `Err`.

Same with `program`.

Reply:
May I ask what I'd need to do in order to "delete" a variable?

Reply:
I think you need to use `glDeleteShader` for the shader objects.

Reply:
okeydokey

Reply:
Should be done now.